### ISSUE TITLE
Voiceover crash when navigating between screens when the content of an underlying view changes 

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#46457]
See internal bug: [b/145575783](http://b/145575783)

On iOS, we encountered a crash when we navigate to a new screen, and then update the UI of the screen beneath the new screen after a small delay.

I managed to create a minimal app to reproduce this crash. To reproduce the crash:
1) Switch on VoiceOver
2) Tap on the '+' button using VoiceOver
3) Tap on the back button using VoiceOver

I am running this app on an iPhone XS running iOS 13.2.3. The following app is built with the latest SDK on https://flutter.dev/docs/get-started/install/macos.

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  final String title;
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  void _incrementCounter() {
    Navigator.push(
      context,
      MaterialPageRoute(builder: (context) => MyHomePage(title: 'Second Page')),
    );
    Future.delayed(const Duration(milliseconds: 1000), () {
      setState(() {
        _counter = 50;
      });
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text('You have pushed the button this many times:'),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.display1,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}
```

**Comment by darrenaustin** [comment]
Thanks for the distilled example. However, I have been unable to reproduce the crash with it on either the master or stable branches of flutter on an iPhone Xs Max running 13.3. I wonder if this was a bug in VoiceOver that was fixed in 13.3. Or perhaps it is hardware specific? I will attempt to get an XS running 13.2.3 and see if I can reproduce it.

In the mean time, is there any more info on the crash itself? Are there any logs or stacktraces available?

**Comment by darrenaustin** [comment]
I have also attempted to reproduce this on an iPhone 8 running 12.4.1 and wasn't able to see the crash.

**Comment by darrenaustin** [comment]
Can we get an output of `flutter doctor` for the case that you are seeing the crash?

**Comment by mehmetf** [comment]
@darrenaustin Please post updates to the internal bug. The customers who filed this do not watch this space. Thanks! (It could be that the crash has been fixed since then)

**Comment by Hixie** [comment]
Current status: we still haven't figured out how to reproduce this.

**Comment by goderbauer** [comment]
It would also be helpful if somebody could post the stacktrace or error message of the crash.

**Comment by Hixie** [comment]
The person who reported this says they can no longer reproduce it, so I'm going to close the bug.
